src/components/radio/radio_dispatcher.ts

```diff
@@ -0,0 +1,27 @@
+import {Injectable} from 'angular2/core';
+
+/**
+ * Class for radio buttons to coordinate unique selection based on name.
+ * Indended to be consumed as an Angular service.
+ * This service is needed because native radio change events are only fired on the item currently
+ * being selected, and we still need to uncheck the previous selection.
+ */
+@Injectable()
+export class MdRadioDispatcher {
+  // TODO(jelbourn): Change this to TypeScript syntax when supported.
+  listeners_: Function[];
+
+  constructor() {
+    this.listeners_ = [];
+  }
+
+  /** Notify other radio buttons that selection for the given name has been set. */
+  notify(name: string) {
+    this.listeners_.forEach(listener => listener(name));
+  }
+
+  /** Listen for future changes to radio button selection. */
+  listen(listener: Function) {
```

Comment:
I _think_ this signature can be made more specific:
```ts
listen(listener: (name: string) => void) { ... }
```
(though I'm not 100% sure if that will transpile to dart)